activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -839,7 +839,15 @@ def connection_pool_list
       end
       alias :connection_pools :connection_pool_list
 
-      def establish_connection(spec)
+      def establish_connection(spec_or_config, name: "primary")
```

Comment:
Think we should leave `name:` a required argument (and make `spec_or_config` optional, if I'm understanding the intent correctly).

Also, I think `spec` and `config` are almost synonyms, which seems to suggest there's something we need to do here. Otherwise that ambiguity bubbles up to users. I'm not sure what that will look like, just saying this makes me a bit uneasy.